### PR TITLE
hb-01 implemented

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+
```
Comment (vicire) [inline pom.xml]: remove empty line

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+public class Main {
+    private static Injector injector = Injector.getInstance("com.dev.cinema");
+
+    public static void main(String[] args) {
+        Movie movie = new Movie();
+        movie.setTitle("Fast and Furious");
+        MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
+        movieService.add(movie);
+
```
Comment (vicire) [inline src/main/java/com/dev/cinema/Main.java]: same here

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (vicire) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: we should not use try with resources in create method coz if the transaction didn't happen, the session would close and we wouldn't go to `catch` and rollback wouldn't happen. So it's better to use try-catch-finally and close session in finally block

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+
```
Comment (vicire) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: remove empty line

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (vicire) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: check [here](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html) point about set

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t insert movie entity: ", e);
```
Comment (vicire) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: better to add your custom exception

File: src/main/java/com/dev/cinema/dao/MovieDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        }
+    }
+
+    public List<Movie> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Movie> getAllmoviesQuery = session.createQuery("from Movie", Movie.class);
+            return getAllmoviesQuery.list();
+        } catch (Exception e) {
+            throw new RuntimeException("Can`t get all movies: ", e);
```
Comment (vicire) [inline src/main/java/com/dev/cinema/dao/MovieDaoImpl.java]: same here

**Review by vicire — CHANGES_REQUESTED** [review]
Remove all redundant empty lines

**Review by vicire — APPROVED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**